### DISCUSSION TITLE
tidyverse: subset vs population?

### DISCUSSION THREAD

**Discussion by dsanders2gilead** [Gilead-Public/gsm#25] [General]
https://github.com/Gilead-BioStats/gsm/blob/2ad58f944650b5a2a548848b63a670cf515bfe03/Sandbox/AssessmentContinuousLongitudinal/Cts_SingleVisit_Analyze.R#L30

Find a way to use tidyverse sytntax to do this instead of for loop.  Something that would possibly look like this example:

```
dfSummary <- dfAlbumin %>%
  group_by(SITEID, AVISITN ) %>%
  select(AVAL, percentCHG) %>%
  summarise(
    mean_aval = mean(AVAL, na.rm = TRUE),
    mean_percentCHG = mean(percentCHG, na.rm = TRUE),
    sample_size_aval = length(!is.na(AVAL)),
    sample_size_pCHG = length(!is.na(percentCHG))
    # sample_size_aval_na = length(AVAL)
  )
```

But instead of doing mean over each group and creating summary df with results, do a t.test with group vs remaining population without group.  The results would still be in the same sort of summary df form as above (i.e a value or output for each subset defined by group_by, example 'SITEID')  but the calculation for each group col (example SITEID) also uses data from the entire population (i.e all SITEID's.  )

**Comment by jwildfire** [comment]
My initial thought is that we want to follow the process using `broom`, `purrr` and friends outlined by Hadley Wickham [here](https://r4ds.had.co.nz/many-models.html) and [here](https://cran.r-project.org/web/packages/broom/vignettes/broom_and_dplyr.html). Basically put everything in a tibble with one row per site and then run the models directly in the tibble. 

We should also keep this [`fct_lump()`](https://forcats.tidyverse.org/reference/fct_lump.html) function in mind for easily creating the desired categorizations on the fly.

**Comment by jwildfire** [comment]
This applies to `Analyze_Wilcoxon()`, so we can look in to it in more detail when we refactor/qualify that assessment.
